### PR TITLE
Validate parameters of randi_range()

### LINKED ISSUE #29688: RandomNumberGenerator::randi_range crashes when to = from - 1
**Godot version:** 3.1.1 stable

**OS/device including version:** N/A

**Issue description:**

I tried to call `randi_range` from the RandomNumberGenerator class by providing parameters such that `to = from - 1` and the project crashes silently. Also, if `to - from < -1`, the generated value will not be inside the given interval.

I created a reproduction project with `from = 4` and `to = 3` triggering a crash. I think the expected behaviour would be to throw an error or even detect and use the parameters to work properly. Also, it should have to do with `ret % (to - from + 1) + from` leading to a division by zero in `random_number_generator.h`.

**Steps to reproduce:**
- Open the reproduction project below and run the main scene

**Minimal reproduction project:**

[repro_randi_range_issue.zip](https://github.com/godotengine/godot/files/3277999/repro_randi_range_issue.zip)

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #29688

### CODE REVIEW

**Comment by groud** [comment]
I believe we should allow `from > to`. This require a simple inequality check. 
The inverted situation only requires to invert the two numbers and to make sure the "from" is still included while the "to" is not.

**Comment by KoBeWi** [comment]
Now is better? Reversing the order produces the same results.

Also
> make sure the "from" is still included while the "to" is not.

Why? The method is inclusive.

**Comment by groud** [comment]
> Why? The method is inclusive.

Ah yes, you are right, I though it was not :)

**Review by groud — APPROVED** [review]

File: core/math/random_number_generator.h [hunk 1]
```diff
@@ -59,7 +59,10 @@ class RandomNumberGenerator : public Reference {
 
 	_FORCE_INLINE_ int randi_range(int from, int to) {
 		unsigned int ret = randbase.rand();
-		return ret % (to - from + 1) + from;
+		if (to < from)
```
Comment (akien-mga) [inline core/math/random_number_generator.h]: Or simpler: `return ret % (Math::abs(to - from) + 1) + to;`

Comment (KoBeWi) [reply core/math/random_number_generator.h]: You mean changing the whole condition into one line? Then you probably need something more like
`return ret % (Math::abs(to - from) + 1) + Math::min(from, to);`
which does one more check.

Comment (akien-mga) [reply core/math/random_number_generator.h]: Indeed, not worth it then.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**